core/opds2_import.py

```diff
@@ -488,29 +488,40 @@ def _extract_media_types_and_drm_scheme_from_link(
 
         media_types_and_drm_scheme = []
 
-        if link.properties:
-            if (
-                not link.properties.availability
-                or link.properties.availability.state
-                == opds2_ast.OPDS2AvailabilityType.AVAILABLE.value
-            ):
-                for acquisition_object in link.properties.indirect_acquisition:
-                    nested_acquisition_object = acquisition_object
-
-                    while nested_acquisition_object.child:
-                        nested_acquisition_object = first_or_default(
-                            acquisition_object.child
-                        )
-
-                    drm_scheme = (
-                        acquisition_object.type
-                        if acquisition_object.type in DeliveryMechanism.KNOWN_DRM_TYPES
-                        else DeliveryMechanism.NO_DRM
-                    )
+        if (
+            link.properties
+            and link.properties.availability
+            and link.properties.availability.state
+            != opds2_ast.OPDS2AvailabilityType.AVAILABLE.value
+        ):
+            self._logger.info(f"Link unavailable. Skipping. {encode(link)}")
+            return []
 
-                    media_types_and_drm_scheme.append(
-                        (nested_acquisition_object.type, drm_scheme)
-                    )
+        # We need to take into account indirect acquisition links
+        if link.properties and link.properties.indirect_acquisition:
+            # We make the assumption that when we have nested indirect acquisition links
```

Comment:
I think it would be useful to mention here that we don't *just* assume this. We verify that we indeed have that pairing for indirect links and, if not, we ignore the link entirely.

Reply:
Added some more verbiage to this comment. Let me know if this addresses your comment, or if you have other suggestions for a comment here.